resources/js/features/forums/components/ShortcodeRenderer/ShortcodeRenderer.test.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  persistedAchievementsAtom,
+  persistedGamesAtom,
+  persistedTicketsAtom,
+  persistedUsersAtom,
+} from '@/features/forums/state/forum.atoms';
+import { render, screen } from '@/test';
+import {
+  createAchievement,
+  createGame,
+  createSystem,
+  createTicket,
+  createUser,
+} from '@/test/factories';
+
+import { ShortcodeRenderer } from './ShortcodeRenderer';
+
+describe('Component: ShortcodeRenderer', () => {
+  it('renders without crashing', () => {
+    // ARRANGE
+    const { container } = render(<ShortcodeRenderer body="" />);
+
+    // ASSERT
+    expect(container).toBeTruthy();
+  });
+
+  it('given a body with a URL tag, renders the URL shortcode component', () => {
+    // ARRANGE
+    const body = '[url]https://example.com[/url]';
+
+    render(<ShortcodeRenderer body={body} />);
+
+    // ASSERT
+    expect(screen.getByTestId('url-embed-https://example.com/')).toBeVisible();
+  });
+
+  it('given a body with a self-closing URL tag, renders the URL shortcode component', () => {
+    // ARRANGE
+    const body = '[url=https://example.com]';
+
+    render(<ShortcodeRenderer body={body} />);
+
+    // ASSERT
+    expect(screen.getByTestId('url-embed-https://example.com/')).toBeVisible();
+  });
+
+  it('given a body with a code tag, renders the code shortcode component', () => {
+    // ARRANGE
+    const body = '[code]this is some stuff inside code tags[/code]';
```

Comment:
Code block now captures the leading whitespace:
```
This change is intended to help enforce the [url=https://docs.retroachievements.org/guidelines/users/global-leaderboard-and-achievement-hunting-rules.html#tl-dr-golden-rules]first golden rule of participating in the global leaderboard[/url]:
[code]
* Using the latest release version of an approved supported emulator
[/code]
```

Before:
![image](https://github.com/user-attachments/assets/fbdca285-72c8-4dc2-81cf-48c5f9b2e1ff)

Now:
![image](https://github.com/user-attachments/assets/1754c63f-4c86-4eff-8277-5031fd53aa89)

Reply:
Good catch. Fixed in latest.